desktop_version/src/Game.cpp

```diff
@@ -228,6 +228,16 @@ void Game::init(void)
 
     customcol=0;
 
+    map.currentregion = 0;
+    for (size_t i = 0; i < SDL_arraysize(map.region); i++)
+    {
+        map.region[i].isvalid = false;
+        map.region[i].rx = 0;
+        map.region[i].ry = 0;
+        map.region[i].rx2 = 0;
+        map.region[i].ry2 = 0;
+    }
+
```

Comment:
1. Why are you initializing `mapclass` attributes inside `Game`?
2. Instead of using a `for`-loop to initialize `region`, I would much rather you write `SDL_zeroa(region);`. It's concise and covers cases in the future when we may add more attributes to `regionstruct`.
3. You need to also zero `map.currentregion` and `map.region` in `scriptclass::hardreset`, otherwise they will persist after loading a new level.